yarn-project/acir-simulator/src/avm/opcodes/external_calls.ts

```diff
@@ -37,34 +36,38 @@ export class Call extends Instruction {
   }
 
   // TODO(https://github.com/AztecProtocol/aztec-packages/issues/3992): there is no concept of remaining / available gas at this moment
-  async execute(machineState: AvmMachineState, journal: AvmJournal): Promise<void> {
-    const callAddress = machineState.memory.getAs<Field>(this.addrOffset);
-    const calldata = machineState.memory.getSlice(this.argsOffset, this.argsSize).map(f => new Fr(f.toBigInt()));
+  async execute(context: AvmContext): Promise<void> {
+    const callAddress = context.machineState.memory.getAs<Field>(this.addrOffset);
+    const calldata = context.machineState.memory
+      .getSlice(this.argsOffset, this.argsSize)
+      .map(f => new Fr(f.toBigInt()));
 
-    const avmContext = AvmContext.prepExternalCallContext(
+    const nestedContext = await AvmContext.createNestedContractCallContext(
```

Comment:
Observe that if you add Call and StaticCall to the instruction set, this use of AvmContext still creates the dependency loop. Also note that this could just be a non-static method and it would make a lot of sense.

With those changes, then you can import AvmContext as type only, and then the loop "disappears".